packages/node/src/integrations/tracing/nest.ts

```diff
@@ -44,7 +54,140 @@ interface MinimalNestJsApp {
 
 const INTEGRATION_NAME = 'Nest';
 
-export const instrumentNest = generateInstrumentOnce(INTEGRATION_NAME, () => new NestInstrumentation());
+const supportedVersions = ['>=8.0.0 <11'];
+
+const sentryPatched = Symbol('sentryPatched');
+
+/**
+ * Represents an injectable target class in NestJS.
+ */
+interface InjectableTarget {
+  name: string;
+  sentryPatched?: symbol;
+  prototype: {
+    use?: (req: unknown, res: unknown, next: (error?: Error | unknown) => void) => void;
+  };
+}
+
+/**
+ * Custom instrumentation for nestjs.
+ *
+ * This hooks into the @Injectable decorator, which is applied on class middleware, interceptors and guards.
+ */
+export class SentryNestInstrumentation extends InstrumentationBase {
+  public static readonly COMPONENT = '@nestjs/common';
+  public static readonly COMMON_ATTRIBUTES = {
+    component: SentryNestInstrumentation.COMPONENT,
+  };
+
+  public constructor(config: InstrumentationConfig = {}) {
+    super('sentry-nestjs', '1.0.0', config); // TODO: instrumentationVersion
+  }
+
+  /**
+   * Initializes the instrumentation by defining the modules to be patched.
+   */
+  public init(): InstrumentationNodeModuleDefinition {
+    const moduleDef = new InstrumentationNodeModuleDefinition(SentryNestInstrumentation.COMPONENT, supportedVersions);
+
+    moduleDef.files.push(this._getInjectableFileInstrumentation(supportedVersions));
+    return moduleDef;
+  }
+
+  /**
+   * Wraps the @Injectable decorator.
+   */
+  private _getInjectableFileInstrumentation(versions: string[]): InstrumentationNodeModuleFile {
+    return new InstrumentationNodeModuleFile(
+      '@nestjs/common/decorators/core/injectable.decorator.js',
```

Comment:
m: Since `Injectable` is exported from the top-level of `@nestjs/common`, I wonder if we can just do `'@nestjs/common'` here :thinking:

Reply:
No doesn't work (changed it locally and it breaks the e2e tests)

Reply:
FWIW they also wrap specific files like this in the nestjs-core instrumentation (https://www.npmjs.com/package/@opentelemetry/instrumentation-nestjs-core)

Reply:
Kinda suck because if they change the file structure this instantly won't work anymore :/ Stuff like this makes us _need_ canary test otherwise we'll never catch breakages like that.

Reply:
Have played around with it but could not make anything else work so far. FWIW I think it's fine here. They have not moved this file in 6 years: https://github.com/nestjs/nest/commits/master/packages/common/decorators/core/injectable.decorator.ts